core/templates/pages/exploration-player-page/layout-directives/content-language-selector.component.ts

```diff
@@ -51,24 +52,30 @@ export class ContentLanguageSelectorComponent implements OnInit {
     private ngbModal: NgbModal,
     private imagePreloaderService: ImagePreloaderService,
     private i18nLanguageCodeService: I18nLanguageCodeService,
-  ) {}
+    private windowRef: WindowRef
+  ) { }
 
   // These properties are initialized using Angular lifecycle hooks
   // and we need to do non-null assertion. For more information, see
   // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
   selectedLanguageCode!: string;
   languageOptions!: ExplorationLanguageInfo[];
   currentGlobalLanguageCode!: string;
+  newLanguageCode!: string;
 
   ngOnInit(): void {
+    const url = new URL(this.windowRef.nativeWindow.location.href);
     this.currentGlobalLanguageCode = (
       this.i18nLanguageCodeService.getCurrentI18nLanguageCode());
     this.selectedLanguageCode = (
       this.contentTranslationLanguageService.getCurrentContentLanguageCode());
     this.languageOptions = (
       this.contentTranslationLanguageService.getLanguageOptionsForDropdown());
+    this.newLanguageCode = (
+      url.searchParams.get('initialContentLanguageCode') ||
+      this.currentGlobalLanguageCode);
```

Comment:
Please add a check for the value for this string.

Reply:
@gp201 what do you mean by check? Do I need to write test for this variable ?

Reply:
Hi! @gp201 I tried to add the tests but since I'm new to the jasmine test , so if i have done something wrong please let me know and  I'd be glad it if you could give me some advice on how to properly write the test for this.

Reply:
@Priyansh61 you wrote the `expect` stmt correctly. Can you combine it with the test above?

Reply:
@gp201 I am confused on how to get urll.searchParam variable in the test, like for the globalCurrentLanguageCode i have mocked the service in the test but I dont know how to do the same for url.searchParam since that includes mocking window service and many others.

Could you please help me fix the above problem ?

Reply:
thanks @gp201 : )